### PR TITLE
Add HTTP(S) proxy support

### PR DESCRIPTION
**Pull request by lee-kode**
This PR adds support for HTTP proxies. Have a read here: https://github.com/PagerDuty/pagerduty-api-python-client/issues/25

### CODE REVIEW

File: pypd/mixins.py [hunk 1]
```diff
@@ -45,6 +46,13 @@ def __init__(self, api_key=None, base_url=None):
             from pypd import base_url
             self.base_url = base_url
 
+        if proxies:
```
Comment (jdcumpson) [inline pypd/mixins.py]: I think this is my only comment for your code - you could probably get as concise here if you used the if statement like:
```python
if not proxies:
    from pypd import proxies
self.proxies = proxies
```

File: README.md [hunk 2]
```diff
@@ -96,6 +96,21 @@ Run all the tests (unittests only currently) with:
 ```sh
 tox
 ```
+## Proxy Support
+There are 2 ways you can use an HTTP(S) proxy. The first method is using environment variables:
```
Comment (jdcumpson) [inline README.md]: Actually another comment here, I would state that this is actually for the `requests` library and their documentation is where to lookup how this works. It's not obvious from this comment that the code will ever use the environment variables (the code doesn't look them up).

File: pypd/mixins.py (hunk 1 above)
Comment (jdcumpson) [inline pypd/mixins.py]: Based on the environment variable discussion above (I realized afterward) - if you explicitly pass a `None` to the requests function for the `proxies` variable, does it still respect the environment variables if set?

Comment (lee-kode) [inline pypd/mixins.py]: **Unit tests** - problem here is that I'm not sure how you'd test a client proxy config without a proxy already setup to tests against. I think we could achieve this mocking? What's your take on it?

**README** - I've fleshed out the proxy section a little bit. Let me know if it's still unclear.

File: README.md (hunk 2 above)
Comment (lee-kode) [inline README.md]: The interrogation of the env variables happens in the _requests_ framework but I did mention this in the README and also provided a link to the docs.

**Pull request merged**